packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -91,11 +115,42 @@ export const Stackbox = () => {
         closeAction={closeTokenPicker}
         initialFocusRef={searchTokenBarRef}
         isOpen={isTokenPickerOpen}
+        onTokenSelect={isPickingTokenFrom ? setTokenFrom : setTokenTo}
       />
       <ConfirmStackModal
         isOpen={isConfirmStackOpen}
         closeAction={closeConfirmStack}
       />
+    </>
+  );
+};
+
+const SelectTokenChip = ({ label, onClick, token }: SelectTokenChipProps) => {
+  const isTokenFrom = label === TOKEN_FROM_LABEL;
+  const handleButtonClick = () => onClick(isTokenFrom);
+
+  return (
+    <div
+      className={cx("flex flex-col space-y-2", {
+        "items-end": !isTokenFrom
+      })}
+    >
+      <BodyText className="text-em-low">{label}</BodyText>
+      {token ? (
+        <ChipButton id={token.address} onClick={handleButtonClick} size="sm">
+          <TokenIcon token={token} />
+          <p className="font-semibold ml-1.5">{token.symbol}</p>
```

Comment:
I think we should use our `BodyText` component.

```suggestion
<BodyText className="ml-1.5">{token.symbol}</BodyText>
```

Reply:
Ahhh agree, seen this a few times and was In doubt but I'd replace every `<p>` tag with our `BodyText` component lol. Will change it here!